packages/repository/src/model.ts

```diff
@@ -106,6 +106,11 @@ export class ModelDefinition {
     const definition = (definitionOrType as PropertyDefinition).type
       ? (definitionOrType as PropertyDefinition)
       : {type: definitionOrType};
+
+    if (definition.id === true && definition.generated === true) {
```

Comment:
To be safe, I am proposing to extend this check to apply `useDefaultIdType: false` only when an explicit `type` is provided.

This property should set `useDefaultIdType: false`:

```ts
@property({
  type: 'string',
  id: true,
  generated: true
})
id: string;
```

This property should NOT set `useDefaultIdType: false`:

```ts
@property({
  id: true,
  generated: true
})
id: string | number;
```

Thoughts?

Reply:
Yes, it seems right to me. I just implemented it